### ISSUE TITLE
quickstart fails

### ISSUE THREAD

**Issue by ftballguy45** [openmaptiles/openmaptiles#1095]
Trying to keep things very simple I run quickstart missouri and the process runs for a while and fails with this error

Creating openmaptiles_generate-vectortiles_run ... done
Generating tiles into /export/tiles.mbtiles for zooms 0..7 inside (data/missouri.bbox is data/missouri.bbox) using 1 threads
Filtering deprecation warnings from the Mapnik's output.
/usr/local/lib/node_modules/tilelive/bin/tilelive-copy:100
        if (err) throw err;
                 ^

Error: bounds must be an array of the form [west, south, east, north]
    at Immediate.<anonymous> (/usr/local/lib/node_modules/tilelive/lib/stream-pyramid.js:71:65)
    at Immediate.immediate._onImmediate (timers.js:577:18)
    at tryOnImmediate (timers.js:534:15)
    at processImmediate [as _immediateCallback] (timers.js:514:5)
1
make: *** [Makefile:405: generate-tiles] Error 1

**Comment by TomPohys** [comment]
Hi @ftballguy45, did you change `BBOX` in `.env`?

**Comment by ftballguy45** [comment]
I might have when I tried to run the process manually as described on the GitHub home page but I had errors there also so I started over, deleting all the docker images created. Do I need to do that before I run quick start?
 
 
  On Tue, Mar 30, 2021 at 1:42 AM, Tomas ***@***.***> wrote:   


Hi @ftballguy45, did you change BBOX in .env?

**Comment by TomPohys** [comment]
No, please do not change it. If in `.env` file will be `BBOX=-180.0,-85.0511,180.0,85.0511` then quickstart will use the BBOX from `*.bbox` file which is calculated based on downloaded area. If you change the BBOX in `.env` it will take value from `.env` file instead of the value in `*.bbox` file.

It is based on https://github.com/openmaptiles/openmaptiles/blob/master/quickstart.sh#L267-L277

**Comment by ftballguy45** [comment]
How can I reset it?
 
 
  On Tue, Mar 30, 2021 at 7:27 AM, Tomas ***@***.***> wrote:   


No, please do not change it. If in .env file will be BBOX=-180.0,-85.0511,180.0,85.0511 then quickstart will use the BBOX from *.bbox file which is calculated based on downloaded area. If you change the BBOX in .env it will take value from .env file instead of the value in *.bbox file.

It is based on https://github.com/openmaptiles/openmaptiles/blob/master/quickstart.sh#L267-L277

**Comment by TomPohys** [comment]
just download the `.env` from this repo. Or be sure that `BBOX=-180.0,-85.0511,180.0,85.0511` is in your `.env` https://github.com/openmaptiles/openmaptiles/blob/master/.env#L19

**Comment by ftballguy45** [comment]
I verified that the .env settings are correct and reran the 
./quickstart.sh missouri

It still fails.

Here is the entire output with the error at the end

[quickstart_missouri_output.txt](https://github.com/openmaptiles/openmaptiles/files/6229548/quickstart_missouri_output.txt)

**Comment by nyurik** [comment]
what's the content of your `data/missouri.bbox` ?

**Comment by ftballguy45** [comment]
How can I find that? Windows guy here and newish to Docker :)
 
 
  On Tue, Mar 30, 2021 at 9:18 AM, Yuri ***@***.***> wrote:   


what's the content of your data/missouri.bbox ?

**Comment by ftballguy45** [comment]
-95.77751,35.99466,-89.0891,40.61664

**Comment by TomPohys** [comment]
There must be something with reading the file in Windows 

Windows
- Generating tiles into /export/tiles.mbtiles for zooms 0..7 inside (data/missouri.bbox is data/missouri.bbox) using 10 threads 
Linux
- Generating tiles into /export/tiles.mbtiles for zooms 0..7 inside (-95.77751,35.99466,-89.0891,40.61664) using 10 threads

So, please, replace `BBOX=-180.0,-85.0511,180.0,85.0511` with `BBOX=-95.77751,35.99466,-89.0891,40.61664` in `.env` file. 

Then, please, run `quickstart.sh missouri` again.

**Comment by ftballguy45** [comment]
I deleted all the images and re-ran it. Same thing

**Comment by nyurik** [comment]
for a comprehensive test, try to rename your `data` dir to something else and let it re-download. I think it somehow got messed up.  I don't think its a docker issue.  That said, running it from Windows could be a challenge at times - but luckily Windows now includes Ubuntu as a subsystem - I wonder if you can simply run it from there instead? (might need to clone/move the repo there).  I personally use Linux and many use Macs, but Windows' ubuntu is pretty good I hear.

**Comment by ftballguy45** [comment]
I'm running docker desktop and using the Linux subsystem 
 
 
  On Tue, Mar 30, 2021 at 12:59 PM, Yuri ***@***.***> wrote:   


for a comprehensive test, try to rename your data dir to something else and let it re-download. I think it somehow got messed up. I don't think its a docker issue. That said, running it from Windows could be a challenge at times - but luckily Windows now includes Ubuntu as a subsystem - I wonder if you can simply run it from there instead? (might need to clone/move the repo there). I personally use Linux and many use Macs, but Windows' ubuntu is pretty good I hear.

**Comment by nyurik** [comment]
let me know if renaming/removing the `data` dir fixed the issue. You may want to try it with a smaller dataset, e.g. `area=belize`

**Comment by ftballguy45** [comment]
Ok will do. Thanks for the help
 
 
  On Tue, Mar 30, 2021 at 1:10 PM, Yuri ***@***.***> wrote:   


let me know if renaming/removing the data dir fixed the issue. You may want to try it with a smaller dataset, e.g. area=belize

**Comment by ftballguy45** [comment]
I renamed the data directory and ran it again. Same result :(

**Comment by ftballguy45** [comment]
I ran it with belize also and same result

**Comment by ftballguy45** [comment]
Any other thoughts or recommendations? This is vanilla out the box setup here

**Comment by ftballguy45** [comment]
I'm surprised that this doesn't work just right out of the box. I'm not doing anything custom or crazy

**Comment by TomPohys** [comment]
I am sorry to hear that, but on Linux and macOS is tested without any problem. I think that Windows user is @zstadler.

Did you try it without quickstart.sh?

**Comment by ftballguy45** [comment]
It's using a Linux container, so it is running on Linux. I ran into issues without the quick start also.
 
 
  On Thu, Apr 8, 2021 at 2:06 PM, Tomas ***@***.***> wrote:   


I am sorry to hear that, but on Linux and macOS is tested without any problem. I think that Windows user is @zstadler.

Did you try it without quickstart.sh?

**Comment by zstadler** [comment]
@ftballguy45,
Docker comes in two flavors for Windows: Docker Desktop runs on Windows Pro and Docker Toolbox runs on other systems. 
What is your environment?

I made several adjustments for running OpenMapTiles with Docker Toolbox using its Docker Quickstart Terminal that provides a bash shell in the Windows host.
Lately I moved to Windows Pro where I run OpenMapTiles from within WSL (Windows Subsystem for Linux) that runs a Ubuntu in a VM.
What is your shell?

**Comment by ftballguy45** [comment]
I'm using Docker Desktop on Windows Pro running Linux containers on WSL2

**Comment by ftballguy45** [comment]
I deleted all volumes, all images, and containers. 
[output.txt](https://github.com/openmaptiles/openmaptiles/files/6288330/output.txt)
Redownloaded this repo and ran 
$ ./quickstart.sh belize
from a cygwin terminal
same error, full output log is attached

**Comment by zstadler** [comment]
@ftballguy45 wrote:
> I'm using Docker Desktop on Windows Pro running Linux containers on WSL2
...
> from a cygwin terminal

When using Docker Desktop on Windows Pro running Linux containers on WSL2 I recommend running OpenMapTiles within the WSL bash window.

I also recommend cloning the OpenMapTiles repo into the Linux filestystem rather than using the Windows filesystem. WSL, and the Docker containers that run within it, have issues accessing the Windows native drives. See [Best Practices](https://docs.docker.com/docker-for-windows/wsl/#best-practices) for Docker Desktop WSL 2 backend.

In other words,
- Work entirely within the WSL2 environment
- Try to avoid using the `/mnt` directory tree
  - Sometimes a `cd .` inside the `/mnt` directory tree can temporarily restore proper access from within the WSL environment

Here is the [quickstart.log](https://github.com/openmaptiles/openmaptiles/files/6289434/quickstart.log) of running `./quickstart.sh belize` in my environment.

**Comment by TomPohys** [comment]
I just finished my test. Here is log for `./quickstart` [quickstart.log](https://github.com/openmaptiles/openmaptiles/files/6289489/quickstart.log)

This was tested on almost absolutely clear MS Windows. Here I write some step by step, how I did it.

### from https://docs.microsoft.com/en-us/windows/wsl/install-win10
#### step 1 (in powershell)
 - dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart

#### step 3
 - dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart

#### restart

#### step 4 install
 - https://wslstorestorage.blob.core.windows.net/wslblob/wsl_update_x64.msi

#### step 5 (in powershell)
 - wsl --set-default-version 2

#### step 6 install Ubuntu 20-04
 - https://www.microsoft.com/store/apps/9n6svws3rx71

#### make sure of enabled CPU virtualization 

#### (optional) windows terminal
 - https://docs.microsoft.com/en-us/windows/wsl/install-win10#install-windows-terminal-optional

#### install docker 
 - download and setup for wsl2
  -  https://docs.docker.com/docker-for-windows/wsl/

#### set uncompressed and unencrypted folder (it already was)
 - USERPROFILE%\AppData\Local\Packages\CanonicalGroupLimited...

#### post-install docker setup
 - https://docs.docker.com/engine/install/linux-postinstall/
 - step 1-4

### in wsl
#### sudo apt install make

#### /home/*user*/$ git clone https://github.com/openmaptiles/openmaptiles
#### /home/*user*/openmaptiles/$ ./quickstart.sh

#### allow firewall access for Docker (pop up window during import-osm)

**Comment by ftballguy45** [comment]
Thank you! I will respond later with more details, but with @TomPohys last comment I was able to get it to run and generate the mbtiles file that I needed.

**Comment by mattmiley** [comment]
Thank you @zstadler & @TomPohys !! I've had a really hard time getting this to work and I had it working a couple of months ago on windows linux subsystem for docker.

You must avoid using the /mnt directory tree!! That was my problem. Do it in your /home/user directory.

Thanks again!!

**Comment by nyurik** [comment]
I think we should add some code to the makefile to catch such cases, warn the user, and abort

**Comment by TomPohys** [comment]
Thanks for the discussion. I will update a readme for Windows users. Discussion can continue in #1103.

Thanks
